Replace SqlDataType AST type with DbDataType in mapping schema

MaceWindu opened the pull request:
- remove (obsolete till v7) use of AST type `SqlDataType` from mapping schema, use `DbDataType`
- remove not needed `DataOptions` parameter from EFC ILinqToDBForEFTools interface methods

In progress/different PR:
- ~remove (obsolete till v7) magic type tables and APIs in `SqlDataType`, types should be configured only in `MappingSchema`~ will be done in separate PR due to complexity

MaceWindu commented on Source/LinqToDB.Scaffold/CodeModel/AST/Basics/AttributeOwner.cs:
```diff
@@ -11,7 +11,7 @@ public abstract class AttributeOwner : ICodeElement
 		protected AttributeOwner(IEnumerable<CodeAttribute>? customAttributes)
 		{
-			_customAttributes = new (customAttributes ?? []);
+			_customAttributes = [.. customAttributes ?? []];
```
Either we disable those rules or fix them as they are now trigger with VS 17.13. I don't mind them in separate PR

viceroypenguin replied:
Let's disable for this PR and reenable and fix in another PR then?

MaceWindu committed:
- remove SqlDataType from MappingSchema

- remove unnecessary DataOptions parameter from ILinqToDBForEFTools EFCore interface

MaceWindu committed:
fix tests

MaceWindu commented:
/azp run test-all

MaceWindu committed:
Merge branch 'master' into refactoring/remove-ast-type-from-mapping-schema

MaceWindu commented:
/azp run test-all

MaceWindu committed:
Update AccessMappingSchema.cs

MaceWindu commented:
/azp run test-all

MaceWindu committed:
Update CachingTests.cs

MaceWindu commented:
/azp run test-all

MaceWindu committed:
Update CachingTests.cs

MaceWindu commented:
/azp run test-all

MaceWindu commented:
/azp run test-all

MaceWindu commented:
This will not make it into v6 due to amount of work required around types. Better to shift it to v7 alongside with planned type mapping refactoring
